### ISSUE TITLE
[P2] Fine-tune classifier with ArcFace metric learning

### ISSUE THREAD

**Issue by thorbearBeining** [christoffersire/nm-ai-2026-ngd#5]
## Problem
Pre-trained EfficientNet-B3 features are general-purpose. Fine-tuning with metric learning loss produces embeddings that are more discriminative for grocery products.

## Solution
1. Prepare training data:
   - Crop annotations from training images (all 356 categories)
   - Add reference images (327 products)
   - Augment: flip, rotation, color jitter, random erasing
2. Fine-tune EfficientNet-B3 with ArcFace loss:
   ```python
   class ArcFaceClassifier(nn.Module):
       backbone = timm.create_model('efficientnet_b3', pretrained=True, num_classes=0)
       projection = nn.Linear(1536, 512)
       arcface = ArcMarginProduct(512, 356, s=30, m=0.5)
   ```
3. Train for 50 epochs with AdamW + cosine annealing
4. Export backbone weights + re-compute reference embeddings

## Depends on: Issue #3 (reference embeddings)

**Comment by christoffersire** [comment]
Closing the post-competition backlog — the competition has ended. Keeping the issue for reference.
